modules/test/playwright/pages/calendar-web/CalendarWidgetPage.ts

```diff
@@ -66,7 +79,28 @@ export class CalendarWidgetPage {
 		await this.allDayCheckbox.hover();
 		await this.allDayCheckbox.setChecked(allDay);
 
+		await this.publishEvent();
+	}
+
+	async publishEvent() {
 		await this.publishEventButton.click();
+		await waitForSuccessAlert(
+			this.page.frameLocator('iframe'),
+			`Success:Your request completed successfully.`
+		);
+	}
+
+	async fillEventWithRecurrence(allDay: boolean, recurrence: Recurrence) {
+		await this.addEventButton.click();
+
+		await this.allDayCheckbox.hover();
+		await this.allDayCheckbox.setChecked(allDay);
+
+		await this.repeatCheckbox.setChecked(true);
+
+		this.modalRecurrencePage = new ModalRecurrencePage(this.page);
```

Comment:
Instead of instantiating `ModalRecurrencePage` here, what you need to do is to click the recurrence button. Define a method like:

```
	async clickRecurrenceButton(name: string = 'Daily') {
		await this.page.frameLocator('iframe').getByRole('link', { name }).click();
	}
```

and call it here: `await this.clickRecurrenceButton()`

Reply:
Also, await next line:
`await this.modalRecurrencePage.addRecurrence(recurrence)`

Reply:
This wasn't necessary because the command `await this.repeatCheckbox.setChecked(true);` opens the modal.